skfda/representation/basis/_custom_basis.py

```diff
@@ -195,6 +195,23 @@ def _evaluate(
     ) -> NDArrayFloat:
         return self.fdata(eval_points)
 
+    def _gram_matrix(self) -> NDArrayFloat:
+        """
+        Compute the Gram matrix.
+
+        Subclasses may override this method for improving computation
+        of the Gram matrix.
+
+        """
+        from ...misc import inner_product_matrix
+
+        if isinstance(self.fdata, FDataBasis):
```

Comment:
Why is this specialized for `FDataBasis`?

Reply:
Leftover from the previous implementation. Fixed now.